src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -1203,8 +1205,73 @@ public bool TryFormat(Span<char> destination, out int charsWritten, [StringSynta
+                        Vector128<byte> highNibbles = UnpackHigh(nibbles, srcVec) & Vector128.Create((byte)0xF);
+                        (Vector128<ushort> v0, Vector128<ushort> v1) = Vector128.Widen(Shuffle(hexMap, lowNibbles));
+                        (Vector128<ushort> v2, Vector128<ushort> v3) = Vector128.Widen(Shuffle(hexMap, highNibbles));
+
+                        // Because of Guid's layout (int _a, short _b, _c, byte ...)
+                        // we have to handle v0 and v1 separately:
+                        v0 = Vector128.Shuffle(v0.AsInt32(), Vector128.Create(3, 2, 1, 0)).AsUInt16();
+                        v1 = Vector128.Shuffle(v1.AsInt32(), Vector128.Create(1, 0, 3, 2)).AsUInt16();
+
+                        ushort* pChar = (ushort*)p;
+                        if (dash)
+                        {
+                            // v0v0v0v0-v1v1-v1v1-v2v2-v2v2v3v3v3v3
+                            v0.Store(pChar + 0);
+                            v1.Store(pChar + 9);
+                            v1 = Vector128.Shuffle(v1.AsInt64(), Vector128.Create(1, 0)).AsUInt16();
+                            v1.Store(pChar + 14);
+                            v2.Store(pChar + 19);
+                            v2 = Vector128.Shuffle(v2.AsInt64(), Vector128.Create(1, 0)).AsUInt16();
+                            v2.Store(pChar + 24);
+                            v3.Store(pChar + 28);
+                            pChar[8] = pChar[13] = pChar[18] = pChar[23] = '-';
+
+                            // We could be smarter here by doing only 5 SIMD stores + permutations
+                            // but extra complexity is not worth it according to benchmarks
+                            p += 36;
+                        }
+                        else
+                        {
+                            // v0v0v0v0v1v1v1v1v2v2v2v2v3v3v3v3
+                            v0.Store(pChar + 0);
+                            v1.Store(pChar + 8);
+                            v2.Store(pChar + 16);
+                            v3.Store(pChar + 24);
+                            p += 32;
+                        }
+
+                        // https://github.com/dotnet/runtime/issues/81609
+                        // VectorTableLookup is not exactly the same but it doesn't matter for the given use case
+                        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+                        static Vector128<byte> Shuffle(Vector128<byte> value, Vector128<byte> mask) =>
+                            Ssse3.IsSupported ? Ssse3.Shuffle(value, mask) : AdvSimd.Arm64.VectorTableLookup(value, mask);
+
+                        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+                        static Vector128<byte> UnpackLow(Vector128<byte> left, Vector128<byte> right) =>
+                            Sse2.IsSupported ? Sse2.UnpackLow(left, right) : AdvSimd.Arm64.ZipLow(left, right);
```

Comment:
nit: I believe the policy to best support crossgen is to ensure there is one throwing path in such helper methods.

This one is likewise common enough that putting it as an internal helper on `Vector128` is likely worthwhile. It's one of the special APIs I plan on bringing up for API review.

Reply:
I already have an upcoming PR where I've got rid of these and @MihaZupan moved Shuffle (ShuffleUnsafe) to Vector128 as internal helper.

As for the crossgen. Presumably it should not even bother precompiling such helpers (in fact it **should not**) - they're private and are always expected to be inlined